### PR TITLE
Issue #1552 feat: course planner integration for events

### PR DESCRIPTION
**Pull request by suvarnakale**

### CODE REVIEW

File: src/components/SessionCardFooter.tsx [hunk 1]
```diff
@@ -136,8 +191,19 @@ const SessionCardFooter: React.FC<SessionCardFooterProps> = ({ item }) => {
         title={item.subject}
         center={'Khapari Dharmu'}
```
Comment (itsvick) [inline src/components/SessionCardFooter.tsx:192]: @suvarnakale hardcoded?

File: src/components/SessionCardFooter.tsx [hunk 2]
```diff
@@ -7,21 +7,76 @@ import CenterSessionModal from './CenterSessionModal';
...
+        const response = await getTargetedSolutions({
+          subject: 'Tamil',
+          class: '4',
+          state: 'Maharashtra',
+          board: 'TQKR',
+          type: 'mainCourse',
+          medium: 'Telugu',
+        });
+
+        const courseData = response.result.data[0];
```
Comment (itsvick) [inline src/components/SessionCardFooter.tsx:46]: @suvarnakale add nullish operator

File: src/components/SessionCardFooter.tsx (hunk 1 above)
Comment (suvarnakale) [reply src/components/SessionCardFooter.tsx:192]: will change it.

File: src/components/SessionCardFooter.tsx [hunk 3]
```diff
@@ -136,8 +191,19 @@ const SessionCardFooter: React.FC<SessionCardFooterProps> = ({ item }) => {
         title={item.subject}
         center={'Khapari Dharmu'}
         date={'25 May, 2024'}
+        primary={'Save'}
```
[the remark this answers is not in the document]
Comment (suvarnakale) [reply src/components/SessionCardFooter.tsx:194]: ok

File: src/components/SessionCardFooter.tsx (hunk 2 above)
Comment (suvarnakale) [reply src/components/SessionCardFooter.tsx:46]: ok

**Pull request merged**